homeassistant/components/igloohome/entity.py

```diff
@@ -0,0 +1,30 @@
+"""Implementation of a base entity that belongs to all igloohome devices."""
+
+from igloohome_api import Api
+
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+
+class IgloohomeBaseEntity(Entity):
+    """A base entity that is a part of all igloohome devices."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, device_id: str, device_name: str, type: str, api: Api) -> None:
```

Comment:
I'd rather pass the entire device. It's more future proof and save on unnecessary parameters.

Reply:
Hey @zweckj , subclasses needs at least the `device_id` to query it's status with the API. Pulling this out of the device seems counter productive as they would need to know how the `DeviceInfo.identifier` is constructed.

Also, including more platforms in the future which would then create subclasses of this entity. Therefore, different entities of the same physical device will easily have matching DeviceInfo objects constructed.

Reply:
yes, but why not pass the entire device, but all those different pieces?  if you pass the entire devices it's cleaner and future platforms have access to any properties of that device

Reply:
I'm not talking about the `DeviceInfo`. The devices your passing is here is your device type from `devices_response = await api.get_devices()`, the `DeviceInfo` is then built in this constructor here, which is fine.

What I don't like is that you are picking properties out of _your_ devices. This would mean if you'd e.g. like to add a serial number to the `DeviceInfo` later, you'd need to add yet another parameter to this function. Instead if you pass and store your device to the entities, you'll be less limited later.

```suggestion
    def __init__(self, device: <your_device_type>, api: Api) -> None:
```

Reply:
Hey @zweckj , thanks for the clarification and can make the changes.

I had imagined that this current structure helps decouple the dependency of the `IgloohomeEntity` from the API's `Device`. 

Should the API change in the future, the changes necessary would then purely fall to just updating the mappings from the new API data structure to the entities constructor, and would not need to change all the internal workings of the entity that had they relied the API's data structure.

Reply:
I'd still think it'd be better to pass the entire thing. If it changes you'll have to update it in `entity.py` and the platforms attributes and not all the constructors all over the place.